support/Pipelines/Bbh/Inspiral.py

```diff
@@ -35,7 +93,45 @@ def inspiral_parameters(
     """
     id_domain_creator = id_input_file["DomainCreator"]["BinaryCompactObject"]
     id_binary = id_input_file["Background"]["Binary"]
-    return {
+    left_object = id_binary["ObjectLeft"]
+    right_object = id_binary["ObjectRight"]
+
+    assert len(left_object) == 1, "Expected only one option for 'ObjectLeft'"
+    assert len(right_object) == 1, "Expected only one option for 'ObjectRight'"
+
+    left_object_type, left_object_params = list(left_object.items())[0]
+    right_object_type, right_object_params = list(right_object.items())[0]
+
+    supported_solutions = [
+        "KerrSchild",
+        "HarmonicSchwarzschild",
+        "Schwarzschild",
+        "SphericalKerrSchild",
+    ]
+    assert left_object_type in supported_solutions, (
+        f"Unknown solution for 'ObjectLeft': '{left_object_type}'. Supported"
+        f" solutions are: {supported_solutions}"
+    )
+    assert right_object_type in supported_solutions, (
+        f"Unknown solution for 'ObjectRight': '{right_object_type}'. Supported"
+        f" solutions are: {supported_solutions}"
+    )
+
+    # ID parameters
+    mass_left = left_object_params["Mass"]
+    mass_right = right_object_params["Mass"]
+    spin_magnitude_left = np.linalg.norm(
+        left_object_params.get("Spin", [0.0, 0.0, 0.0])
+    )
+    spin_magnitude_right = np.linalg.norm(
+        right_object_params.get("Spin", [0.0, 0.0, 0.0])
+    )
```

Comment:
I'm not sure where you are using the spins, but these should be the true spins from a horizon find, not the bare, conformal spins

Reply:
They are used for some control system parameters above. But yes I agree, this is included with using the correct masses from the horizon finds.